### PR TITLE
Assert that links in related.rst are a list

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

### CODE REVIEW

File: doc/test_messages_documentation.py [hunk 1]
```diff
@@ -46,6 +49,12 @@ def get_functional_test_files_from_directory(input_dir: Path) -> List[Tuple[str,
                 suite.append(
                     (message_dir.stem, message_dir / "bad.py"),
                 )
+            if (message_dir / "related.rst").exists():
+                with open(message_dir / "related.rst", encoding="utf-8") as file:
+                    text = file.read()
+                    assert text.startswith(
+                        "-"
```
Comment (Pierre-Sassoulas) [inline doc/test_messages_documentation.py:56]:
```suggestion
                        "*"
```

File: doc/test_messages_documentation.py [hunk 2]
```diff
@@ -46,6 +49,12 @@ def get_functional_test_files_from_directory(input_dir: Path) -> List[Tuple[str,
...
+                    ), f"{message_dir / 'related.rst'} should be a list using '-'."
```
Comment (Pierre-Sassoulas) [inline doc/test_messages_documentation.py:57]:
```suggestion
                    ), f"{message_dir / 'related.rst'} should be a list using '*'."
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
Are you sure that rst lists are done with ``-`` and not ``*`` ? https://gdal.org/contributing/rst_style.html#lists

**Comment by DanielNoord** [comment]
> Are you sure that rst lists are done with `-` and not `*` ? https://gdal.org/contributing/rst_style.html#lists

See:
https://pylint--6213.org.readthedocs.build/en/6213/messages/warning/cell-var-from-loop.html

Seems to work? I tested there were more cases with `-` currently so this seemed easier 😄

**Comment by Pierre-Sassoulas** [comment]
See for example the current state of cell-var-from-loop : https://pylint.pycqa.org/en/latest/messages/warning/cell-var-from-loop.html
It seems to also work using ``-`` https://pylint--6213.org.readthedocs.build/en/6213/messages/warning/cell-var-from-loop.html

**Comment by DanielNoord** [comment]
Btw, I really like the new theme 😄 Think it was a good discussion to go ahead with it!

File: doc/test_messages_documentation.py (hunk 2 above)
Comment (Pierre-Sassoulas) [inline doc/test_messages_documentation.py:57]:
```suggestion
                    assert text.startswith(                        "-"                    ) or text.startswith(                        "*"                    ), f"{message_dir / 'related.rst'} should be a list using '-'."
```

Comment (DanielNoord) [reply doc/test_messages_documentation.py:57]: Why not require consistency now? We're not too far in to change it now

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
I don't think we should force the use of one or the other if it's valid rst. (Especially as it's not automatically formatted).

**Comment by DanielNoord** [comment]
> I don't think we should force the use of one or the other if it's valid rst. (Especially as it's not automatically formatted).

If we do force consistency we can more easily parse this info programmatically. I have been thinking of making a PR for `pylint-errors` to get the info they use form our documentation instead of their internal database to make sure the effort of writing these examples is not duplicated. Removing inconsistencies like these would help with doing so 😄

**Comment by Pierre-Sassoulas** [comment]
> we do force consistency we can more easily parse this info programmatically.

Hmm right.

>  I have been thinking of making a PR for pylint-errors to get the info they use form our documentation instead of their internal database

I think vald-phoenix is currently busy "fighting for the bright future of 🇺🇦" so it make take some time to merge it but also isn't the base idea to integrate the work and then to not have to maintain pylint-error anymore ?

**Review by Pierre-Sassoulas — APPROVED** [review]
Let's normalize then.

**Comment by DanielNoord** [comment]
> I think vald-phoenix is currently busy "fighting for the bright future of 🇺🇦" so it make take some time to merge it but also isn't the base idea to integrate the work and then to not have to maintain pylint-error anymore ?

We could also do that. However, they also have a CLI tool which we would need to integrate. Not sure if we have the bandwidth to maintain that.
Making `pylint-errors` a CLI tool that depends on our documentation as a `git submodule` might be a good compromise, but that's definitely something for "future us". Right now we're not even on par with their data yet.

**Comment by DanielNoord** [comment]
> > Making pylint-errors a CLI tool that depends on our documentation as a git submodule might be a good compromise, but that's definitely something for "future us". Right now we're not even on par with their data yet.
> 
> I think we have a --msg-help option already, right ?

Ah of course. We could probably integrate the code examples there, although that would inflate the size of the `pylint` package. Something to think about...

**Pull request merged**